stdlib/src/collections/list.mojo

```diff
@@ -111,8 +127,19 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
         Args:
             capacity: The requested capacity of the list.
         """
-        self.data = UnsafePointer[T].alloc(capacity)
         self.size = 0
+        self._small_buffer = Self._small_buffer_type(uninitialized=True)
+
+        @parameter
+        if Self.sbo_enabled:
+            if capacity <= Self._small_buffer_size:
+                self.capacity = Self._small_buffer_size
+                # needed to avoid "potential indirect access to uninitialized value 'self.data'"
+                self.data = UnsafePointer[T]()
+                self.data = self._small_buffer.unsafe_ptr()
+                return
+
+        self.data = UnsafePointer[T].alloc(capacity)
         self.capacity = capacity
```

Comment:
Nit: Consider implementing the fbstring sbo strategy to increase the amount of characters we store in the object before spilling to the heap.

https://youtu.be/kPR8h4-qZdk?feature=shared&t=527

Reply:
That's a very smart trick! If that's okay by you we can leave it for another pull request, as to avoid too much complexity in a single PR.